lib/webapp.js

```diff
@@ -233,21 +233,40 @@ const staticHandlers = {
         but this should be enforced somehow in the future.
       */
       if (!type) {
-        do404(req.url, res, "A plugin type must be specified");
-        return;
+        type = "all"
       }
       const acceptLanguage = 
         translationUtils.getAcceptLanguageFromCookies(req.cookies) || req.headers['accept-language'] || '';
       const pluginDefs = plugins.map(p => p.exportTranslatedDef(acceptLanguage));
+            
+      let filteredPluginDefs = []
+      let allowedPlugins;
+
+      if (options.serverConfig.dataserviceAuthentication.rbac) {
+        if (req.username) {
+          allowedPlugins = configService.getAllowedPlugins(configService.CONFIG_SCOPE_USER, options, req.username);
+        } else {
+          allowedPlugins = configService.getAllowedPlugins(configService.CONFIG_SCOPE_PRODUCT, options);
+        }
+        for(let plugin of pluginDefs) {
+          let obj = allowedPlugins.allowedPlugins.find(o => o.identifier === plugin.identifier)
```

Comment:
It's possible that the above configService returns allowedPlugins = null.
If so, you should return the unfiltered pluginDefs object.

Reply:
Also, your filter ignores the plugin version, which is in the JSON you get.
So, right now you are not allowed to have more than one version of the same plugin in the server, but you should at least add this logic so that we dont forget to do it later... so, if it is "*", then it always succeeds, but otherwise... try the 'semver' library we already have in this codebase... see here: https://github.com/zowe/zlux-server-framework/blob/master/lib/depgraph.js#L294